p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -440,46 +440,51 @@ public void onConnection(Connection connection) {
-                                    // We might get connection drops which are not caused by the node going offline, so
-                                    // we give more tolerance with that approach, giving the node the change to
-                                    // refresh the TTL with a refresh message.
-                                    // We observed those issues during stress tests, but it might have been caused by the
-                                    // test set up (many nodes/connections over 1 router)
-                                    // TODO investigate what causes the disconnections.
-                                    // Usually the are: SOCKET_TIMEOUT ,TERMINATED (EOFException)
-                                    protectedStorageEntry.backDate();
-                                    if (protectedStorageEntry.isExpired(this.clock)) {
-                                        log.info("We found an expired data entry which we have already back dated. " +
-                                                "We remove the protectedStoragePayload:\n\t" + Utilities.toTruncatedString(protectedStorageEntry.getProtectedStoragePayload(), 100));
-                                        removeFromMapAndDataStore(protectedStorageEntry, hashOfPayload);
-                                    }
-                                } else {
-                                    log.debug("Remove data ignored as we don't have an entry for that data.");
-                                }
-                            }
-                        }
-                    });
-        }
+        if (closeConnectionReason.isIntended)
+            return;
+
+        if (!connection.getPeersNodeAddressOptional().isPresent())
+            return;
+
+        NodeAddress peersNodeAddress = connection.getPeersNodeAddressOptional().get();
+
+        // Retrieve all the eligible payloads based on the node that disconnected
+        ArrayList<Map.Entry<ByteArray, ProtectedStorageEntry>> toBackDate =
+                map.entrySet().stream()
+                        .filter(entry -> entry.getValue().getProtectedStoragePayload() instanceof ExpirablePayload)
+                        .filter(entry -> entry.getValue().getProtectedStoragePayload() instanceof RequiresOwnerIsOnlinePayload)
+                        .filter(entry -> ((RequiresOwnerIsOnlinePayload) entry.getValue().getProtectedStoragePayload()).getOwnerNodeAddress().equals(peersNodeAddress))
+                        .collect(Collectors.toCollection(ArrayList::new));
+
+        // Backdate each payload
+        toBackDate.forEach(mapEntry -> {
+            // We only set the data back by half of the TTL and remove the data only if is has
+            // expired after that back dating.
+            // We might get connection drops which are not caused by the node going offline, so
+            // we give more tolerance with that approach, giving the node the change to
```

Comment:
NIT: Small typo in the original code: `change` should be `chance`

Reply:
Addressed in e8c8225